test/e2e/federated-service.go

```diff
@@ -223,28 +233,23 @@ func waitForFederatedServiceShard(cs *release_1_3.Clientset, namespace string, s
 		// Renaming for clarity/readability
 		clSvc := clSvcList.Items[0]
 
-		// The federation service has no cluster IP.  Clear any cluster IP before
-		// comparison.
-		clSvc.Spec.ClusterIP = ""
-
 		Expect(clSvc.Name).To(Equal(service.Name))
 		// Some fields are expected to be different, so make them the same before checking equality.
 		clSvc.Spec.ClusterIP = service.Spec.ClusterIP
 		clSvc.Spec.ExternalIPs = service.Spec.ExternalIPs
 		clSvc.Spec.DeprecatedPublicIPs = service.Spec.DeprecatedPublicIPs
 		clSvc.Spec.LoadBalancerIP = service.Spec.LoadBalancerIP
 		clSvc.Spec.LoadBalancerSourceRanges = service.Spec.LoadBalancerSourceRanges
+		for i := range clSvc.Spec.Ports {
```

Comment:
minor nit: Prefer "for i, port := ....".  Then you can just use "port.NodePort below.